### ISSUE TITLE
Support gnome.compile_resources with generated resources files

### ISSUE THREAD

**Issue by baedert** [mesonbuild/meson#706]
I'm trying to generate a `.gresource.xml` file for `gnome.compile_resources` in my `meson.build` using a `configure_file`, but passing the `configure_file` to `gnome.compile_resources` results in the `.gresource.xml` file not being found. The strange part is that the generated file is correctly placed in the builddir (with correct name and contents) but `glib-compile-resources` still can't find it.

**Comment by baedert** [comment]
This is how I currently implemented this:

```
glsl_sources = [
  'gl3-texture-2d.fs.glsl',
  'gl3-texture-2d.vs.glsl',
  'gl3-texture-rect.fs.glsl',
  'gl3-texture-rect.vs.glsl',
  'gl2-texture-2d.fs.glsl',
  'gl2-texture-2d.fs.glsl',
  'gl2-texture-rect.vs.glsl',
  'gl2-texture-rect.vs.glsl',
  'gles2-texture.fs.glsl',
  'gles2-texture.vs.glsl',
]
resource_xml = ''
foreach glsl_file : glsl_sources
  resource_xml += '    <file alias="glsl/@0@">resources/glsl/@0@</file>\n'.format(glsl_file)
endforeach


cdata = configuration_data()
cdata.set('GDK_GRESOURCE_XML_FILES', resource_xml)
gdkresources_xml = configure_file(
  input  : 'gdk.gresource.xml.meson',
  output : 'gdk.gresource.xml',
  configuration : cdata
)

gdkresources = gnome.compile_resources(
  'gdkresources',
  gdkresources_xml,
  source_dir: meson.build_root(),
  # source_dir: '.',
  c_name: '_gdk',
  extra_args: '--manual-register'
)
```
The `source_dir` part is just a test, none of the options I tried seem to be working.

**Comment by nirbheek** [comment]
This was fixed in dabf0c1882772700fed2c6c18ca06919a8bc280d (0.38.1 and 0.39.0), but I forgot about this issue.
